src/EditorFeatures/Core/Extensibility/Completion/ICustomCommitCompletionProvider.cs

```diff
@@ -16,6 +14,6 @@ namespace Microsoft.CodeAnalysis.Editor
     /// </summary>
     internal interface ICustomCommitCompletionProvider
     {
-        void Commit(CompletionItem completionItem, ITextView textView, ITextBuffer subjectBuffer, ITextSnapshot triggerSnapshot, char? commitChar);
+        void Commit(CompletionItem completionItem, Document document, ITextView textView, ITextBuffer subjectBuffer, ITextSnapshot triggerSnapshot, char? commitChar);
```

Comment:
slight waryness as i believe we support completion in non-Document cases.  However, it's likely we don't support *Custom completion* there, so this is likely fine.

Reply:
All of the callers already have a `Document` in scope. This is an internal interface that could be adjusted in the future if we need to support a new scenario.

Reply:
➡️ Left this one as-is